### DISCUSSION TITLE
Put description for shared Datasets

### DISCUSSION THREAD

**Discussion by kossa** [pestphp/pest#630] [Q&A] [answered]
I have an array with a lot of info to make a test, for that I put it in shared datasets, my issue is I want to set a description, dont want to show all array like this

<img width="1362" alt="image" src="https://user-images.githubusercontent.com/1175584/209842691-dd1f371f-0707-4269-a67f-b5b64a59348b.png">

**Comment by fabio-ivona** [comment, marked as answer]
you should be able to add a key to your dataset entries, so they can be used as description:

```php
dataset('my dataset', [
    'item 1' => '......',
    'item 2' => '......',
]);
```

**Reply by kossa** [reply]
Alright it works fine

**Reply by kossa** [reply]
It works for simple key=>value but not for value as array, it returns only the first item, try this

Test file:
```
test('All works fine', function ($data) {
    dump($data);
})->with('my dataset');
```

Dataset file:
```
dataset('my dataset', [
    'data1' => ['name' => 'name1', 'email' => 'email@gmail.com'],
    'data2' => ['name' => 'name2', 'email' => 'email2@gmail.com'],
]);
```

**Reply by fabio-ivona** [reply]
that's because you are selecting only the first element of an array of data for each dataset entry

you should do either:

```
test('All works fine', function ($name, $email) {
    dump($data);
})->with('my dataset');
```

Dataset file:
```
dataset('my dataset', [
    'data1' => ['name' => 'name1', 'email' => 'email@gmail.com'],
    'data2' => ['name' => 'name2', 'email' => 'email2@gmail.com'],
]);
```

or

```
test('All works fine', function ($data) {
    dump($data);
})->with('my dataset');
```

Dataset file:
```
dataset('my dataset', [
    'data1' => [['name' => 'name1', 'email' => 'email@gmail.com']],   //<---- note the double square brackets
    'data2' => [['name' => 'name2', 'email' => 'email2@gmail.com']],
]);
```

**Reply by kossa** [reply]
It works perfectly, I liked the second solution
